cmd/snap-bootstrap/cmd_initramfs_mounts_test.go

```diff
@@ -2477,6 +2478,626 @@ func (s *initramfsMountsSuite) TestInitramfsMountsRecoverModeHappyEncrypted(c *C
 	c.Assert(filepath.Join(dirs.SnapBootstrapRunDir, fmt.Sprintf("%s-model-measured", s.sysLabel)), testutil.FilePresent)
 }
 
+func checkDegradedJSON(c *C, exp map[string]interface{}) {
+	b, err := ioutil.ReadFile(filepath.Join(boot.InitramfsHostUbuntuDataDir, "degraded.json"))
+	c.Assert(err, IsNil)
+	degradedJSONObj := make(map[string]interface{}, 0)
+	err = json.Unmarshal(b, &degradedJSONObj)
+	c.Assert(err, IsNil)
+
+	c.Assert(degradedJSONObj, DeepEquals, exp)
+}
+
+func (s *initramfsMountsSuite) TestInitramfsMountsRecoverModeEncryptedDegradedFallbackDataHappy(c *C) {
+	s.mockProcCmdlineContent(c, "snapd_recovery_mode=recover snapd_recovery_system="+s.sysLabel)
```

Comment:
It looks like the setup is always the same for all the new tests, maybe we could do that in SetUpTest?

Reply:
mmm to do this we would need to split the suites into a recover suite, an install suite, a run suite, etc. but this could be done. that would also make for a more logical split of all the tests in this file into multiple files which would be nice since we are now sitting at over 3000 lines here